Mono.Cecil/ModuleKind.cs

```diff
@@ -32,6 +32,7 @@ public enum TargetArchitecture {
 		ARM = 0x01c0,
 		ARMv7 = 0x01c4,
 		ARM64 = 0xaa64,
+		RiscV64 = 0x5064,
```

Comment:
Why do you need to define the enum value when it is no longer used in the rest Cecil?

It is ok to have unnamed enum values.

Reply:
It is checked in the linker here: https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs#L79. Perhaps we can remove that validation and cleanup the enum?

Reply:
Yep, I have just made the same suggestion above.

Reply:
Thank you so much. I have a question.
If `CalculateArchitecture` is removed, I think the enum values are not needed for .NET runtime.
As far as I know, cecil is used in other many codes as well as .NET runtime. I am not sure but some codes can use the enum values. Maybe I think it is hard to remove  the enum values in `jbevain/cecil`.
Is it okay to have different implementation from the original cecil?

Reply:
Cecil uses it to determine Windows native image offsets, while linker is using it for Windows PE machine type, which is irrelevant in .NET Core. I think returning 0 for new architectures from https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs#L128 (while keeping the existing ones for .NET Framework compat) is more desirable.

See how relaxed the readers are implemented: https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs#L144 and similar discussion in https://github.com/dotnet/runtime/issues/77697.

---

I think this patch +/- test adjustment will do it for now:
```diff
--- a/runtime/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs
+++ b/runtime/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs
@@ -76,7 +76,8 @@ TargetArchitecture CalculateArchitecture (TargetArchitecture readyToRunArch)
                        if (architectureMap.TryGetValue ((ushort) readyToRunArch, out TargetArchitecture pureILArch)) {
                                return pureILArch;
                        }
-                       throw new BadImageFormatException ("unrecognized module attributes");
+
+                       return 0; // Unknown architecture which ultimately translates to AnyCPU in .NET Core
                }
 
                protected override bool ConditionToProcess ()
```

Reply:
I would go even further, delete the CalculateArchitecture method, and change [this line](https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs#L128) to `module.Architecture = 0;  // Unknown architecture which ultimately translates to AnyCPU`

Reply:
@jkotas @am11 I think I misunderstood something.
My plan was `Add AnyCPU to TargetArchitecutre enum in cecil` (with enum value or just enum name) and then set `TargetArchitecture.AnyCPU` to `module.Architecutre` in the line of runtime code.

> I would go even further, delete the CalculateArchitecture method, and change [this line](https://github.com/dotnet/runtime/blob/fa1164c327052042120bf0d3b6fc81e86cfab69d/src/tools/illink/src/linker/Linker.Steps/OutputStep.cs#L128) to module.Architecture = 0;  // Unknown architecture which ultimately translates to AnyCPU

I think it is good. I made a PR. https://github.com/dotnet/runtime/pull/101038
I will close cecil PRs.
Thank you!